### PR TITLE
fix(ingest): proper null skip logic in serialization

### PR DESCRIPTION
**Pull request by hsheth2**
Makes the null removal schema-aware.

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/cli/check_cli.py [hunk 1]
```diff
@@ -22,6 +26,33 @@ def mce_file(json_file: str) -> None:
     click.echo(report)
 
 
+@check.command(hidden=True)
+@click.argument("json-file", type=click.Path(exists=True, dir_okay=False))
+def canonicalize_mcp_file(json_file: str) -> None:
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/cli/check_cli.py:31]: how about adding this as a flag to the existing check mce_file command?
maybe we should call it a different name in the light of mce -> mcp:
`datahub check metadata_file --json-file <metadata_events.json> --lintFix`

File: metadata-ingestion/src/datahub/ingestion/run/pipeline.py [hunk 2]
```diff
@@ -160,7 +161,7 @@ def __init__(
             sink_config = self.config.sink.dict().get("config") or {}
             self.sink: Sink = sink_class.create(sink_config, self.ctx)
             logger.debug(f"Sink type:{self.config.sink.type},{sink_class} configured")
-            logger.info(f"Sink configured successfully. {self.sink.configured()}")
+            logger.debug(f"Sink configured successfully. {self.sink.configured()}")
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/run/pipeline.py:164]: I think this is a pretty important log line. Only gets printed once in the execution of a recipe. Any particular reason to drop it to debug?

File: metadata-ingestion/src/datahub/ingestion/sink/file.py [hunk 3]
```diff
@@ -59,14 +58,18 @@ def write_record_async(
...
-        obj = remove_empties(record.to_obj())
+        obj = record.to_obj()
 
         if self.wrote_something:
             self.file.write(",\n")
 
         json.dump(obj, self.file, indent=4)
         self.wrote_something = True
 
+        # record_string = str(record_envelope.record)
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/sink/file.py:69]: seems like a remnant of reverting my previous commit. remove these?

**Review by shirshanka — CHANGES_REQUESTED** [review]
Seems like a few extra changes from the subject of the PR, intentional?

**Comment by hsheth2** [comment]
@shirshanka I'll split the unrelated changes into a separate PR - this one should be good to go now.

**Review by shirshanka — APPROVED** [review]
LGTM

**Pull request merged**